homestar-runtime/src/event_handler/swarm_event.rs

```diff
@@ -580,6 +598,27 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
                 "peer connection closed, cause: {cause:?}"
             );
             event_handler.connected_peers.remove_entry(&peer_id);
+
+            // Remove peer from DHT if not in configured peers
+            if event_handler.node_addresses.iter().all(|multiaddr| {
```

Comment:
Could this made more succinct withs something like a filter_map?

Reply:
We would also need a `fold` to evaluate over the mapped results, so it doesn't seem like it would be more succinct. Plus the `all` is a bit more expressive that we are evaluating over boolean conditions.